ai_ta_backend/vector_database.py

```diff
@@ -1201,6 +1216,73 @@ def format_for_json(self, found_docs: List[Document]) -> List[Dict]:
 
     return contexts
   
-  
+  def check_for_duplicates(self, texts: List[Dict], metadatas: List[Dict[str, Any]]) -> bool:
+    """
+    For given metadata, fetch docs from Supabase based on S3 path or URL. 
+    If docs exists, concatenate the texts and compare with current texts, if same, return True.
+    """
+    print("in check_for_duplicates")
+    
+    doc_table = os.getenv('NEW_NEW_NEWNEW_MATERIALS_SUPABASE_TABLE')
+    course_name = metadatas[0]['course_name']
+    incoming_s3_path = metadatas[0]['s3_path']
+    url = metadatas[0]['url']
+    original_filename = incoming_s3_path.split('/')[-1][37:] # remove the 37-char uuid prefix
+    print("Extracted filename from incoming s3_path: ", original_filename)
+
+    # check if uuid exists in s3_path

```

Comment:
@KastanDay I added this UUID check (specific to the `uuid.uuid4()` pattern) using regex. Let me know if its good.

Reply:
Ok I checked this and it's perfect, assuming we always stick to uuidv4, which we will. 

ChatGPT says :
> This regex is specifically tailored for version 4 UUIDs, which have particular characteristics in their structure:
The 13th character (part of the third group of digits) is always 4, denoting the version.
The 17th character (first character of the fourth group of digits) is one of 8, 9, a, or b, which are used to indicate the variant of the UUID.